### PR TITLE
Added CustomSSOCredentialsProvider to retain higher-level ClientConfiguration

### PR DESCRIPTION
**Pull request by bhavya2109sharma**
*Issue #, if available:*

*Description of changes:*

The existing `SSOCredentialsProvider` class was overwriting the higher-level `ClientConfiguration` settings, which is leading to issue while authenticating through SSO.Added a new custom provider class called `CustomSSOCredentialsProvider` that inherits from `SSOCredentialsProvider` and allows the use of a custom `ClientConfiguration`.

 In the `Reload()` method of `CustomSSOCredentialsProvider`, a new `ClientConfiguration` object is created using the provided `m_clientConfig` member variable, and the `scheme` and `region` properties are set before passing it to the `SSOCredentialsClient` constructor.

This will ensure  that the higher-level `ClientConfiguration` is retained, and the `SSOCredentialsClient` is created with the correct configuration. 

*Check all that applies:*
- [x] Did a review by yourself.
- [] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [] Checked if this PR is a breaking (APIs have been changed) change.
- [] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [x] Linux
- [x] Windows
- [ ] Android
- [x] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
Duplicate of #2860, which avoids manipulating the `ClientConfiguration`.

File: src/aws-cpp-sdk-core/include/aws/core/auth/CustomSSOCredentialsProvider.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                config.scheme = Aws::Http::Scheme::HTTPS;
+                config.region = m_ssoRegion;
+
+                m_client = Aws::MakeUnique<Aws::Internal::SSOCredentialsClient>(SSO_CREDENTIALS_PROVIDER_LOG_TAG, config);
+
+                SSOCredentialsProvider::Reload();
+            }
+
+        private:
+            Aws::Client::ClientConfiguration m_clientConfig;
```
Comment (teo-tsirpanis) [inline src/aws-cpp-sdk-core/include/aws/core/auth/CustomSSOCredentialsProvider.h:31]: There is already a `ClientConfiguration` in the base SSO credentials provider. Is the duplication intentional?

File: src/aws-cpp-sdk-core/include/aws/core/auth/CustomSSOCredentialsProvider.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+#pragma once
+
+#include <aws/core/auth/SSOCredentialsProvider.h>
+#include <aws/core/client/ClientConfiguration.h>
+
+namespace Aws {
+    namespace Auth {
+        class AWS_CORE_API CustomSSOCredentialsProvider : public SSOCredentialsProvider {
```
Comment (sbiscigl) [inline src/aws-cpp-sdk-core/include/aws/core/auth/CustomSSOCredentialsProvider.h:13]: this class isnt needed we should just be adding a constructor the `SSOCredentialsProvider` itself, no need to create a indirection.

**Comment by sbiscigl** [comment]
> Duplicate of #2860, which avoids manipulating the `ClientConfiguration`.

This is right we should be merging @teo-tsirpanis 's PR instead. Sorry @teo-tsirpanis for not getting to your PR sooner. going to a do a review of it today, from skimming it looks fine though and accomplishes what we want to do. Lets jump over to a conversation on that.

@bhavya2109sharma closing this one out and we can work on merging the original .

**Comment by sbiscigl** [comment]
@teo-tsirpanis left a few comments on your PR, if addressed we can prioritize getting it in.

**Pull request closed without merging**